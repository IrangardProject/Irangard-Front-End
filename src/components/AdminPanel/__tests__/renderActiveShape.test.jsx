import React from 'react';
import { shallow } from 'enzyme';
import { Sector } from 'recharts';
import { renderActiveShape, data5 } from '../option5';

describe('renderActiveShape', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<Sector />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render correctly', () => {
        const wrapper = shallow(<Sector />);
        expect(wrapper).toMatchSnapshot();
    }
    );
    test('should render the correct elements and props', () => {
        const props = {
          cx: 100,
          cy: 100,
          midAngle: 90,
          innerRadius: 20,
          outerRadius: 40,
          startAngle: 0,
          endAngle: 180,
          fill: 'red',
          payload: { name: 'Group A' },
          percent: 0.5,
          value: 400,
        };
        });
      });
      
      describe('data5', () => {
        test('should contain the correct array data', () => {
          expect(data5).toEqual([
            { name: 'Group A', value: 400 },
            { name: 'Group B', value: 300 },
            { name: 'Group C', value: 300 },
            { name: 'Group D', value: 200 },
          ]);
        });
});