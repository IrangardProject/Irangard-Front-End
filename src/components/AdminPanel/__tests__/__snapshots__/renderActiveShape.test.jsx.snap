// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderActiveShape should render correctly 1`] = `ShallowWrapper {}`;

exports[`renderActiveShape should render correctly 2`] = `ShallowWrapper {}`;
